src/main/java/com/google/devtools/build/lib/rules/java/JavaCompileAction.java

```diff
@@ -175,6 +153,20 @@ public JavaCompileAction(
     this.dependencyArtifacts = dependencyArtifacts;
     this.outputDepsProto = outputDepsProto;
     this.classpathMode = classpathMode;
+    FileSystem fileSystem = null;
+    try {
+      fileSystem = new JavaIoFileSystem();
+    } catch (DigestHashFunction.DefaultHashFunctionNotSetException e) {
+      e.printStackTrace();
+    }
+    if(fileSystem != null)
+      this.diagnosticsFile = new Artifact.SourceArtifact(
```

Comment:
I'm quite sure this is not the best place to do this, but since I did not find a better place, I'd love to hear your feedback regarding it